packages/sdk/src/oracle/services/statemachines/utils.ts

```diff
@@ -11,12 +19,15 @@ export const Step = <P = undefined, M extends Memory = undefined>(handlers: Hand
   assert(handlers[context.state], "No handler for state: " + context.state);
   // conext is read only for some reason, have to copy it
   const copy = Object.assign({}, context);
+
   try {
+    // remove sleep, assume it should be reset if needed
+    if (copy.sleep) copy.sleep = undefined;
     const handler = handlers[copy.state];
     // have to copy this out because memory becomes read only for some reason?
     const memory = Object.assign({}, copy.memory);
     // handlers take in params and return a new state or nothin
-    const state = await handler(copy.params, memory);
+    const state = await handler(copy.params, memory, new ContextClient(copy));
```

Comment:
adding this contextclient just enables updates to context from within handler. in this case it can set "sleep" for next tick, putting this at the back of the statemachine queue